template/runtime/src/lib.rs

```diff
@@ -468,14 +468,16 @@ impl_runtime_apis! {
 				_ => {
 					let dispatch_info = xt.get_dispatch_info();
 
-					// TODO How do I get the tip?
-					let tip = 0;
+					let fee_details = TransactionPayment::query_fee_details(
+						xt.clone(),
+						xt.encode().len() as u32
+					);
```

Comment:
I don't think we will get the real tip value this way because of https://github.com/paritytech/substrate/blob/master/frame/transaction-payment/src/lib.rs#L357

It just passes in 0 as the tip.

Yesterday I started looking down this path https://github.com/paritytech/substrate/blob/master/frame/transaction-payment/src/lib.rs#L538

Reply:
Sorry for hurrying it in! I was totally convinced that was it. Please feel free to roll it back.